apps/analysis_framework/migrations/0041_widget_mapping.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 3.2.17 on 2024-02-22 05:02
+
+from django.db import migrations,transaction
+
+@transaction.atomic
+def analysis_framework_widgets_mapping(apps, schema_editor):
+    AnalysisFramework = apps.get_model('analysis_framework', 'AnalysisFramework')
+    analysis_framework_data = AnalysisFramework.objects.filter(properties__isnull=False)
+    for af in analysis_framework_data:
+        if af.properties != {}:
+            widget_properties = af.properties
+            widget_properties['stats_config']['widget_1d'] = widget_properties['stats_config'].get('widget1d') or widget_properties['stats_config'].get('widget_1d') or []  # noqa
```

Comment:
```suggestion
            widget_properties_stats_properties['widget_1d'] = {k: widget_1d[k] for k in widget_1d if k in widget_properties_stats_properties}
```

Reply:
@sudan45 this can be efficient when there are more numbers of keys to be matched.